clients/repo_client.go

```diff
@@ -30,6 +30,7 @@ const HeadSHA = "HEAD"
 // RepoClient interface is used by Scorecard checks to access a repo.
 type RepoClient interface {
 	InitRepo(repo Repo, commitSHA string, commitDepth int) error
+	NewClient(repo string, commitSHA string, commitDepth int) (RepoClient, error)
```

Comment:
I'm not sure I understand the usecase for `NewClient` API. Could we not use `CreateGitHubRepoClient` instead?